unstructured_ingest/v2/processes/connectors/couchbase.py

```diff
@@ -0,0 +1,152 @@
+import json
+from dataclasses import dataclass, field
+from datetime import timedelta
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Optional
+
+from unstructured_ingest.enhanced_dataclass import enhanced_field
+from unstructured_ingest.utils.data_prep import batch_generator
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    UploadContent,
+    Uploader,
+    UploaderConfig,
+    UploadStager,
+    UploadStagerConfig,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import (
+    DestinationRegistryEntry,
+)
+
+if TYPE_CHECKING:
+    from couchbase.cluster import Cluster
+
+CONNECTOR_TYPE = "couchbase"
+SERVER_API_VERSION = "1"
+
+
+@dataclass
+class CouchbaseAccessConfig(AccessConfig):
+    connection_string: Optional[str] = None
+    username: Optional[str] = None
+    password: str = enhanced_field(sensitive=True, default=None)
+
+    def __post_init__(self):
+        if not self.connection_string and not (self.username and self.password):
+            raise ValueError("Either connection string or username and password must be set")
```

Comment:
Now that I understand the use of these fields, since password is the only sensitive value, it's the only thing that should live in `CouchbaseAccessConfig`, with no `__post_init__` require since it's not marked `Optional`. Let's move `username` and `connection_string` to `CouchbaseConnectionConfig`.
```suggestion
@dataclass
class CouchbaseAccessConfig(AccessConfig):
    password: str
```

Reply:
Makes sense. So I have made all the fields as required (Because they are required to tell exactly where to store data)